### PR TITLE
Fix state of tracks in browser app

### PR DESCRIPTION
**Pull request by imran-ebi**
This PR fixes the browser tracks not being saved when navigated away from the browser app. Also, the following are persisted via web storage:

- Track list tabs
- Track list groups

TODO:

- [ ] Write tests

The related JIRA ticket is: https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-83

### CODE REVIEW

File: src/ensembl/src/content/app/browser/Browser.tsx [hunk 1]
```diff
@@ -3,7 +3,8 @@ import React, {
   useCallback,
   useRef,
   useEffect,
-  Fragment
+  Fragment,
```
Comment (azangru) [inline src/ensembl/src/content/app/browser/Browser.tsx]: Not that it's important, but can we please remove the fragment from jsx below? It doesn't seem to be doing anything useful here.

**Comment by azangru** [comment]
If I am reading the code correctly, it seems to assume that there is only one species that is ever going to be shown by the genome browser. However, In our case of several (multiple) species, should the genome browser have the same track settings for all of these species (as I suspect will be the effect of this code)?

File: src/ensembl/src/content/app/browser/track-panel/track-panel-list/TrackPanelList.tsx [hunk 2]
```diff
@@ -103,14 +108,31 @@ const TrackPanelList: FunctionComponent<TrackPanelListProps> = (
     };
   };
 
-  const getTrackListItem = (track: TrackPanelItem | null) => {
+  const getDefaultTrackStatus = (categoryName: string, trackName: string) => {
+    const statesOfCategory = props.trackStates[categoryName];
+
+    let defaultTrackStatus = ImageButtonStatus.ACTIVE;
```
Comment (azangru) [inline src/ensembl/src/content/app/browser/track-panel/track-panel-list/TrackPanelList.tsx]: Very confused why a track status is defined by an image button status.

File: src/ensembl/src/content/app/browser/browser-storage-service.test.ts [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+      );
+      const result = browserStorageService.getTrackListToggleStates();
+
+      expect(result).toEqual({});
+
+      mockStorageService.get.mockRestore();
+    });
+  });
+
+  describe('.updateTrackListTogglestates()', () => {
```
Comment (azangru) [inline src/ensembl/src/content/app/browser/browser-storage-service.test.ts]: capital s in States

**Comment by imran-ebi** [comment]
>If I am reading the code correctly, it seems to assume that there is only one species that is ever going to be shown by the genome browser. However, In our case of several (multiple) species, should the genome browser have the same track settings for all of these species (as I suspect will be the effect of this code)?

This code will not be covering it. I will create another PR for it as it is part of a different task that needs more work. This PR, however, is to fix the state of tracks.

**Comment by azangru** [comment]
But why do we want to even touch this if it will need immediate rework? It’s not like we are publishing this to the live site until the end of June? Or are we?

**Comment by imran-ebi** [comment]
> But why do we want to even touch this if it will need immediate rework? It’s not like we are publishing this to the live site until the end of June? Or are we?

We are publishing this by end of June and Dan requires this to work on the genome browser to show tracks based on the saved track states.

File: src/ensembl/src/content/app/browser/track-panel/track-panel-list/TrackPanelList.tsx (hunk 2 above)
Comment (imran-ebi) [reply src/ensembl/src/content/app/browser/track-panel/track-panel-list/TrackPanelList.tsx]: That's what was used previously and I thought of keeping it for now. But I understand what you mean. This is semantically wrong and I probably need to create a new enum for the track states.

**Comment by ens-st3** [comment]
Can you add a link to this to the relevant JIRA @imran-ebi ?

**Comment by imran-ebi** [comment]
As discussed with @azangru offline, a new branch will be created based on this branch to include species related state as well. So will be closing this PR.

**Pull request closed without merging**